tools/tpm2_policycountertimer.c

```diff
@@ -0,0 +1,271 @@
+     */
+    if (size > sizeof(ctx.operand_b.buffer)) {
+        LOG_ERR("Size is too large for TPM2B_OPERAND. Got %u, max size is: %zu",
+                size, sizeof(ctx.operand_b.buffer));
+        return false;
+    }
+
+    ctx.operand_b.size = size;
+    memcpy(ctx.operand_b.buffer, &data.b, size);
+
+    return true;
+}
+
+static bool on_arg(int argc, char **argv) {
+
+    if (argc > 1) {
+        LOG_ERR("Specify one argument as value/ parameter=value.");
+        return false;
+    }
+
+    if (!strcmp("safe", argv[0])) {
+        ctx.offset = OFFSET_TPMS_TIME_INFO_SAFE;
+        ctx.operand_b.size = 1;
+        ctx.operand_b.buffer[0] = TPM2_YES;
+        return true;
+    }
+
+    const char *value;
+    const char *key;
+    char *a = argv[0];
+    char *split = strchr(a, '=');
+    if (!split) {
+      value = argv[0];
+      key = "time";
+    } else {
+        split[0] = '\0';
+        value = split + 1;
+        key = a;
+    }
+
+    if (!value[0]) {
+      LOG_ERR("Must specify a corresponding value");
+      return false;
+    }
+
+  // look up key and process value
+    if (!strcmp("time", key)) {
+        return convert_keyvalue_to_operand_buffer(value,
+            OFFSET_TPMS_TIME_INFO_TIME, sizeof(uint64_t));
+    }
+
+    if (!strcmp("clock", key)) {
+        return convert_keyvalue_to_operand_buffer(value,
+            OFFSET_TPMS_TIME_INFO_CLOCK, sizeof(uint64_t));
+    }
+
+    if (!strcmp("resets", key)) {
+        return convert_keyvalue_to_operand_buffer(value,
+            OFFSET_TPMS_TIME_INFO_RESETS, sizeof(uint32_t));
+    }
+
+    if (!strcmp("restarts", key)) {
+        return convert_keyvalue_to_operand_buffer(value,
+            OFFSET_TPMS_TIME_INFO_RESTARTS, sizeof(uint32_t));
+    }
+
+    LOG_ERR("Unknown argument. Specify time/ clock/ resets/ restarts/ safe");
+    return false;
+}
+
+static bool on_option(char key, char *value) {
+
+    bool result = true;
+
+    switch (key) {
+    case 'L':
+        ctx.policy_digest_path = value;
+        break;
+    case 'S':
+        ctx.session_path = value;
+        break;
+    case 0:
+        ctx.operation = TPM2_EO_EQ;
+        break;
+    case 1:
+        ctx.operation = TPM2_EO_NEQ;
+        break;
+    case 2:
+        ctx.operation = TPM2_EO_SIGNED_GT;
+        break;
+    case 3:
+        ctx.operation = TPM2_EO_UNSIGNED_GT;
+        break;
+    case 4:
+        ctx.operation = TPM2_EO_SIGNED_LT;
+        break;
+    case 5:
+        ctx.operation = TPM2_EO_UNSIGNED_LT;
+        break;
+    case 6:
+        ctx.operation = TPM2_EO_SIGNED_GE;
+        break;
+    case 7:
+        ctx.operation = TPM2_EO_UNSIGNED_GE;
+        break;
+    case 8:
+        ctx.operation = TPM2_EO_SIGNED_LE;
+        break;
+    case 9:
+        ctx.operation = TPM2_EO_UNSIGNED_LE;
+        break;
+    case 10:
+        ctx.operation = TPM2_EO_BITSET;
+        break;
+    case 11:
+        ctx.operation = TPM2_EO_BITCLEAR;
+        break;
+    default:
```

Comment:
If you specify multiple options, the tool works respecting the last option. I would make all the option cases fall through and use the number to index into an array of operations for the assignment and then set a flag that option handked. And check that flag in this handler that it's not set. If it's set issue an error.